xbmc/cores/VideoPlayer/DVDMessage.cpp

```diff
@@ -88,11 +88,11 @@ bool CDVDMsgGeneralSynchronize::Wait(unsigned int milliseconds, unsigned int sou
   return true;
 }
 
-void CDVDMsgGeneralSynchronize::Wait(volatile bool *abort, unsigned int source)
+void CDVDMsgGeneralSynchronize::Wait(std::atomic<bool>& abort, unsigned int source)
 {
   while(!Wait(100, source))
   {
-    if(abort && *abort)
+    if(abort && abort)
```

Comment:
look again here. the old code checked the ptr for existence and then its value - this is not needed with the reference here. Remove one of the abort